aws/resource_aws_ec2_managed_prefix_list_entry.go

```diff
@@ -0,0 +1,424 @@
+
+	err = resource.Retry(5*time.Minute, func() *resource.RetryError {
+
+		entries, err = getEc2ManagedPrefixListEntries(conn, getEntriesInput)
+
+		if err != nil {
+			return resource.NonRetryableError(fmt.Errorf("error listing EC2 Managed Prefix List (%s) entries: %s", d.Id(), err))
+		}
+
+		cidr := findEntryMatch(entry, entries)
+		if cidr == nil {
+			log.Printf("[DEBUG] Unable to find matching entry (%s) for EC2 Managed Prefix List %s",
+				id, pl_id)
+			return resource.RetryableError(fmt.Errorf("No match found"))
+		}
+
+		log.Printf("[DEBUG] Found entry for EC2 Managed Prefix List entry (%s): %s", id, cidr)
+		return nil
+	})
+	if isResourceTimeoutError(err) {
+		entries, err = getEc2ManagedPrefixListEntries(conn, getEntriesInput)
+
+		cidr := findEntryMatch(entry, entries)
+		if cidr == nil {
+			return fmt.Errorf("Error finding matching EC2 Managed Prefix List entry: %s", err)
+		}
+	}
+	if err != nil {
+		return fmt.Errorf("Error finding matching entry (%s) for EC2 Managed Prefix List %s", id, pl_id)
+	}
+
+	d.SetId(id)
+	return resourceAwsEc2ManagedPrefixListEntryRead(d, meta)
+}
+
+func resourceAwsEc2ManagedPrefixListEntryRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	pl_id := d.Get("prefix_list_id").(string)
+	pl, err := finder.ManagedPrefixListByID(conn, pl_id)
+	if _, notFound := err.(managedPrefixListNotFound); notFound {
+		// The EC2 Managed Prefix List containing this entry no longer exists.
+		d.SetId("")
+		return nil
+	}
+	if err != nil {
+		return fmt.Errorf("Error finding EC2 Managed Prefix List (%s) for entry (%s): %s", pl_id, d.Id(), err)
+	}
+
+	input := &ec2.GetManagedPrefixListEntriesInput{
+		PrefixListId: pl.PrefixListId,
+	}
+	var entry *ec2.PrefixListEntry
+	var entries []*ec2.PrefixListEntry
+	entries, err = getEc2ManagedPrefixListEntries(conn, input)
+
+	if err != nil {
+		return fmt.Errorf("error listing entries of EC2 Managed Prefix List (%s): %w", d.Id(), err)
+	}
+
+	log.Printf("[DEBUG] Entries %v", entries)
+
+	p, err := expandPrefixEntry(d, pl)
+	if err != nil {
+		return err
+	}
+
+	if len(entries) == 0 {
+		log.Printf("[WARN] No entries were found for EC2 Managed Prefix List (%s) looking for entry (%s)",
+			*pl.PrefixListName, d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	entry = findEntryMatch(p, entries)
+
+	if entry == nil {
+		log.Printf("[DEBUG] Unable to find matching entry (%s) for EC2 Managed Prefix List %s",
+			d.Id(), pl_id)
+		d.SetId("")
+		return nil
+	}
+
+	log.Printf("[DEBUG] Found entry for EC2 Managed Prefix List (%s): %s", pl_id, entry)
+
+	setFromPrefixEntry(d, p)
+
+	if strings.Contains(d.Id(), "_") {
+		// import so fix the id
+		id := prefixListEntryIDHash(pl_id, p)
+		d.SetId(id)
+	}
+
+	d.Set(
+		"version",
+		pl.Version,
+	)
+
+	return nil
+}
+
+func resourceAwsEc2ManagedPrefixListEntryUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+
+	if d.HasChange("description") {
```

Comment:
Would it be easier and cause the same user disruption to make `description` `ForceNew`?